### PR TITLE
OLD Create USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet

### PR DESCRIPTION
**Pull request by ajgriebel**
Create TRENDS_YEAR.jsonnet file for yearly areas.

### CODE REVIEW

File: catalog/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+    interval: 1,
+  },
+  summaries: {
+    gsd: [30],
+    // TODO(schwehr): Use units.thing for gee:units 
+    'eo:bands': [
+      {
+        name: 'annual_herbaceous_break_point',
+        description: 'Presence or absence of structural breaks in each component for each year',
+        'gee:units': 'units_counts',
```
Comment (schwehr) [inline catalog/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet:120]: Don't quote the right hand side of the `:`.   Those are variables.

`'units_counts'` → `units_counts`

e.g.

`'gee:units': units_counts,`

File: catalog/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+      '1985-01-01T00:00:00Z',
+      '2021-12-31T00:00:00Z'),
+  'gee:interval': {
+    type: 'cadence',
+    unit: 'year',
+    interval: 1,
+  },
+  summaries: {
+    gsd: [30],
+    // TODO(schwehr): Use units.thing for gee:units
```
Comment (schwehr) [inline catalog/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet:115]: Remove this todo line

File: examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Import the NLCD RCMAP TRENDS YEAR collection.
+var dataset = ee.ImageCollection('projects/ee-rcmap/assets/RCMAP_V5_TRENDS/YEAR');
+
+// Filter the collection to the 2019 product.
+var nlcd2019 = dataset.filter(ee.Filter.eq('system:index', '2019')).first();
+
+// Each product has multiple bands for different rangeland categories.
+print('Bands:', nlcd2019.bandNames());
```
Comment (schwehr) [inline examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js:8]: Delete the print line

File: examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// Import the NLCD RCMAP TRENDS YEAR collection.
+var dataset = ee.ImageCollection('projects/ee-rcmap/assets/RCMAP_V5_TRENDS/YEAR');
+
+// Filter the collection to the 2019 product.
+var nlcd2019 = dataset.filter(ee.Filter.eq('system:index', '2019')).first();
```
Comment (schwehr) [inline examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js:5]: Use `ee.Filter.date`.  Make sure you test this as I haven't.

```
var year = '2019';
var nlcd2019 = dataset.filter(ee.Filter.date(ee.Date(year).getRange('year')));
```

e.g.

https://github.com/google/earthengine-catalog/blob/main/examples/NOAA/NOAA_NHC_HURDAT2_atlantic.js#L5

**Review by schwehr — CHANGES_REQUESTED** [review]

File: examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js (hunk 3 above)
Comment (ajgriebel) [reply examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js]: Done

File: examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js (hunk 4 above)
Comment (ajgriebel) [reply examples/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.js]: Updated, tested and worked

**Comment by schwehr** [comment]
All the errors are units related and I can handle them on the Google side.  So do not worry about these:

```
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'annual_herbaceous_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'bare_ground_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'herbaceous_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'litter_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'sagebrush_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'shrub_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/20[19](https://github.com/google/earthengine-catalog/actions/runs/4574893259/jobs/8077070306?pr=451#step:8:20)_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'non_sagebrush_shrub_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/[20](https://github.com/google/earthengine-catalog/actions/runs/4574893259/jobs/8077070306?pr=451#step:8:21)19_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'perennial_herbaceous_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'tree_break_point gee:units not known: 1 (count)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'annual_herbaceous_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'bare_ground_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'herbaceous_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'litter_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'sagebrush_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'shrub_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'non_sagebrush_shrub_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'perennial_herbaceous_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'tree_segment_pvalue gee:units not known: 1 (confidence)', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'annual_herbaceous_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'bare_ground_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'herbaceous_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'litter_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'sagebrush_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'shrub_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'non_sagebrush_shrub_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'perennial_herbaceous_segment_slope gee:units not known: c%/a', IssueLevel.error)
Issue('USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR', 'USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.json', 'eo_extension', 'tree_segment_slope gee:units not known: c%/a', IssueLevel.error)
```

File: catalog/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet [hunk 5]
```diff
@@ -0,0 +1,533 @@
+local id = 'USGS/NLCD_RELEASES/2019_REL/RCMAP/V5/TRENDS/YEAR';
+local subdir = 'USGS';
+
```
Comment (schwehr) [inline catalog/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet:3]: If you can, add `local version = '5';` similar to how you did on the other PR.

If not, we can always do that after the PR.

Comment (ajgriebel) [reply catalog/USGS/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_TRENDS_YEAR.jsonnet:3]: Line 37 has the "version: '5'," (same as the non-year Trends).  I'm not seeing a "local version = '5' in the above section for either. We can look at it when you get back.

**Pull request closed without merging**